File: 2020/02/02.js
let fs = require('fs');

const isPasswordValid = data => {
    const [policy, password] = data.split(":").map(x => x.trim());
    const [numbers, letter] = policy.split(" ");
    const [num1, num2] = numbers.split("-").map(x => +x - 1);
    return password[num1] === letter ^ password[num2] === letter;
}

const fileData = fs.readFileSync('./input', {encoding:'utf8', flag:'r'});
const lines = fileData.split("\n");

const numValidPasswords = lines.map(x => isPasswordValid(x) ? 1 : 0)
                               .reduce((accumulator, x) => accumulator + x);

console.log("Number of valid passwords: " + numValidPasswords);

File: 2020/02/02.py
with open("input") as file:
    data = [line for line in file]

num_valid = 0

for line in data:
    parts = line.split()
    numbers = parts[0].split('-')
    index1 = int(numbers[0]) - 1
    index2 = int(numbers[1]) - 1

    letter = parts[1][0]
    password = parts[2]

    if (password[index1] == letter) != (password[index2] == letter):
        num_valid += 1

print(num_valid)